crates/ascii-streamer/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+#[macro_use]
+extern crate tracing;
+
+use std::{future::Future, sync::Arc};
+
+use tokio::{
+    io::{AsyncBufReadExt, BufReader},
+    sync::Notify,
+};
+
+pub trait AsciiStreamSocket {
+    fn send(&mut self, value: String) -> impl Future<Output = anyhow::Result<()>> + Send;
+    fn close(&mut self) -> impl Future<Output = ()> + Send;
+}
+
+#[tracing::instrument(skip_all)]
+pub async fn ascii_stream(
+    mut websocket: impl AsciiStreamSocket,
+    input_stream: impl tokio::io::AsyncRead + Unpin, // A file usually
```

Comment:
suggestion: Use a docstring if you want to elaborate on that, but this comment is not very useful otherwise.

```suggestion
    input_stream: impl tokio::io::AsyncRead + Unpin,
```